proto/sf/ethereum/type/v2/type.proto

```diff
@@ -34,8 +34,23 @@ message Block {
   // reward for the block mined, the transfer of ETH to the miner happens outside the normal
   // transaction flow of the chain and is recorded as a `BalanceChange` here since we cannot
   // attached it to any transaction.
+  //
+  // Only available in DetailLevel: EXTENDED
   repeated BalanceChange balance_changes = 11;
 
+  enum DetailLevel{
+    DETAILLEVEL_EXTENDED = 0;
+    // DETAILLEVEL_TRACE = 1; // TBD
+    DETAILLEVEL_BASE = 2;
+  }
+
+  // DetailLevel affects the data available in this block.
+  //
+  // EXTENDED describes the most complete block, with traces, balance changes, storage changes. It is extracted during the execution of the block.
+  // BASE describes a block that contains only the block header, transaction receipts and event logs: everything that can be extracted using the base JSON-RPC interface (https://ethereum.org/en/developers/docs/apis/json-rpc/#json-rpc-methods) 
+  //      Furthermore, the eth_getTransactionReceipt call has been avoided because it brings only minimal improvements at the cost of requiring an archive node or a full node with complete transaction index.
+  DetailLevel detail_level = 12;
```

Comment:
missing description for TRACE (oh I see.. it's not there yet, but we'll add when it arrives.

Reply:
exactly, we don't know what is covered exactly with trace, let's not waste time doing this preemptively.